tests/ValidateSbrpTests/ValidateSbrpTests.cs

```diff
@@ -0,0 +1,170 @@
+
+    private string RepoRoot { get; set; }
+
+    public ITestOutputHelper output { get; set; }
+
+    public ValidateSbrpTests(ITestOutputHelper _output)
+    {
+        output = _output;
+
+        RepoRoot = Environment.CurrentDirectory.Substring(0, Environment.CurrentDirectory.IndexOf("artifacts"));
+
+        string buildPackagesDirectory = Path.Combine(RepoRoot, "artifacts/source-build/self/src/artifacts/packages/Release/Shipping");
+
+        if (!Directory.Exists(buildPackagesDirectory))
+        {
+            throw new DirectoryNotFoundException($"Directory {buildPackagesDirectory} does not exist. Try building with './build.sh -sb' first.");
+        }
+
+        Packages = Directory.GetFiles(buildPackagesDirectory, "*.nupkg", SearchOption.AllDirectories);
+    }
+
+    [Fact]
+    public void CheckForSbrpAttribute()
+    {
+        string[] targetPacks = Directory.GetDirectories(Path.Combine(RepoRoot, "src/targetPacks/ILsrc")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
+        string[] textOnlyPackages = Directory.GetDirectories(Path.Combine(RepoRoot, "src/textOnlyPackages/src")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
+
+        var filteredPackages = new List<string>();
+        
+        foreach (var package in Packages)
+        {
+            string packageName = Path.GetFileNameWithoutExtension(package).ToLower();
+            packageName = Regex.Replace(packageName, VerionPattern, string.Empty);
+            if (!targetPacks.Contains(packageName) && !textOnlyPackages.Contains(packageName))
+            {
+                filteredPackages.Add(package);
+            }
+        }
+
+        foreach (var package in filteredPackages)
+        {
+            string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName(), Path.GetFileNameWithoutExtension(package));
+
+            try
+            {
+                Directory.CreateDirectory(tempDirectory);
+
+                ZipFile.ExtractToDirectory(package, tempDirectory);
+
+                var dlls = Directory.GetFiles(tempDirectory, "*.dll", SearchOption.AllDirectories);
+
+                foreach (var dll in dlls)
+                {
+                    using var stream = new FileStream(dll, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                    using var peReader = new PEReader(stream);
+
+                    MetadataReader reader = peReader.GetMetadataReader();
+
+                    Assert.True(HasSbrpAttribute(reader), $"{package}/{Path.GetRelativePath(tempDirectory, dll)} does not contain the {SbrpAttributeType} attribute with key='source' and value='source-build-reference-packages'.");
+                }
+            }
+            finally
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+    }
+
+    [Fact]
+    public async Task CheckForSignature()
+    {
+        var trustProviders = new List<ISignatureVerificationProvider>()
```

Comment:
Consider utilizing an array here versus the overhead of instantiating a list.  It also helps simplify the code a little.

```
ISignatureVerificationProvider[] trustProviders = [new SignatureTrustAndValidityVerificationProvider()];
var verifier = new PackageSignatureVerifier(trustProviders);
var settings = SignedPackageVerifierSettings.GetDefault();
```

Reply:
Fixed in https://github.com/dotnet/source-build-reference-packages/commit/ce1a70dc071c6a5adce2b2d0ca6285031ee12258